cmd/korg/audit.go

```diff
@@ -0,0 +1,375 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"sort"
+	"strings"
+
+	"github.com/olekukonko/tablewriter"
+)
+
+type Contribution struct {
+	Rank         int
+	Username     string
+	ContribCount int
+	Orgs         []string
+}
+
+type Values struct {
+	Items [][]interface{} `json:"values,omitempty"`
+}
+
+type Frames struct {
+	Schema map[string]interface{} `json:"schema,omitempty"`
+	Data   Values                 `json:"data,omitempty"`
+}
+
+type DevStatsRequest struct {
+	Queries []Query `json:"queries"`
+}
+
+type Query struct {
+	RefID        string `json:"refId"`
+	DatasourceID int    `json:"datasourceId"`
+	RawSQL       string `json:"rawSql"`
+	Format       string `json:"format"`
+}
+
+type AuditOptions struct {
+	Period            string
+	ActivityThreshold int
+	OutputFile        string
+	ExceptionsFile    string
+	CheckOwners       bool
+	CheckTeams        bool
+}
+
+type UserInfo struct {
+	Username      string
+	Contributions int
+	Orgs          []string
+	Teams         map[string][]string
+	IsOwner       bool
+}
+
+type Exception struct {
+	Username string
+	Reason   string
+}
+
+func GetAllUsersInOrgs(o Options, orgs []string) (map[string]UserInfo, error) {
+	users := make(map[string]UserInfo)
+	config, err := LoadOrgs(o)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, org := range orgs {
+		for _, u := range append(config[org].Members, config[org].Admins...) {
+			if user, found := users[strings.ToLower(u)]; found {
+				if user.Orgs == nil {
+					user.Orgs = []string{}
+				}
```

Comment:
Do we need this branch? Or is it just in there as a sanity check? Because we'll never really hit this iiuc.

Reply:
this is just a sanity check.